### PR TITLE
Improve contribution docs

### PR DESCRIPTION
**Pull request by davidmehren**
### Component/Part
Contribution documentation

### Description
This PR improves our contribution docs by adding an overview on how somebody can help the project. It also clarifies the steps required to make a PR.

### Steps

- [x] added implementation
- [x] added / updated tests
- [x] added / updated documentation
- [x] extended changelog

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,42 +1,68 @@
-# Contributing
+# Contributing to HedgeDoc
 
-When contributing to this repository, please first discuss the change you wish to make via issue,
-email, or any other method with the owners of this repository before making a change.
+Thanks for your help in improving the HedgeDoc project!
 
-Please note we have a [code of conduct](CODE-OF-CONDUCT.md), please follow it in all your
+Please note we have a [code of conduct][code-of-conduct], please follow it in all your
```
Comment (InnayTool) [inline CONTRIBUTING.md:5]: maybe we should use _comply with_

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,42 +1,68 @@
...
+
+### Do you intend to add a new feature or change an existing one?
+
+* Suggest your idea via a new GitHub issue. After a confirmation about your idea, you can start writing code. Our maintainers and other project developers can provide useful details about the architecture and show you relevant issues and discussions.
+
+### Do you want to work on translations?
+
+* If you want to improve a translation or add a new translation altogether, we handle those via [POEditor](poeditor).
+
+HedgeDoc is a volunteer effort. We encourage you to pitch in and to help us making this project even better.
```
Comment (DerMolly) [inline CONTRIBUTING.md:33]: this seems like a starting sentence and not an ending sentence…

**Review by DerMolly — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,42 +1,68 @@
...
-their contribution under the project's license.
-Please read [docs/legal/developer-certificate-of-origin.txt][dcofile].
 If you can certify it, then just add a line to every git commit message:
 
-```git
+```
   Signed-off-by: Random J Developer <random@developer.example.org>
 ```
 
 Use your real name (sorry, no pseudonyms or anonymous contributions).
```
Comment (InnayTool) [inline CONTRIBUTING.md:49]: this line could be a problem...

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,42 +1,68 @@
...
-`git ci` and the commit will be signed.
+
+If you set your `user.name` and `user.email` git configs, you can sign your commit automatically with `git commit -s`.
+You can also use git [aliases](https://git-scm.com/book/tr/v2/Git-Basics-Git-Aliases) like `git config --global alias.ci 'commit -s'`.
+Now you can commit with `git ci` and the commit will be signed.
+
+## Submitting a Pull Request
+
+1. Submit an issue describing your proposed change.
+   We will try to respond to your issue promptly.
```
Comment (InnayTool) [inline CONTRIBUTING.md:58]: _promptly_ sounds like there were an 24/7-hotline ^^

File: CONTRIBUTING.md (hunk 3 above)
Comment (DerMolly) [reply CONTRIBUTING.md:49]: How so?

File: CONTRIBUTING.md (hunk 2 above)
Comment (davidmehren) [reply CONTRIBUTING.md]: I moved this statement to the beginning of the file

File: CONTRIBUTING.md (hunk 4 above)
Comment (davidmehren) [reply CONTRIBUTING.md:57]: We could also write "as soon as possible" but I'm not sure if that makes such a big difference :D

File: CONTRIBUTING.md (hunk 3 above)
Comment (InnayTool) [reply CONTRIBUTING.md:48]: I think there are many people who don't want to add their real name.

File: CONTRIBUTING.md (hunk 4 above)
Comment (InnayTool) [reply CONTRIBUTING.md:57]: I would use "as soon as possible".

File: CONTRIBUTING.md (hunk 3 above)
Comment (DerMolly) [reply CONTRIBUTING.md:48]: Well we can't really help with that, now can we?

**Review by DerMolly — APPROVED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (davidmehren) [reply CONTRIBUTING.md:48]: Contributors need to use their real name for legal reasons and this PR also didn't change this line.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,42 +1,67 @@
...
-their contribution under the project's license.
-Please read [docs/legal/developer-certificate-of-origin.txt][dcofile].
 If you can certify it, then just add a line to every git commit message:
 
-```git
+```
   Signed-off-by: Random J Developer <random@developer.example.org>
 ```
 
 Use your real name (sorry, no pseudonyms or anonymous contributions).
```
Comment (mrdrogdrog) [inline CONTRIBUTING.md:48]: "Please use your real name, mr. random developer". Can we replace the example name with a more serious one?

**Review by mrdrogdrog — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (davidmehren) [reply CONTRIBUTING.md]: I really don't like "comply" but now use "respect" instead.

File: CONTRIBUTING.md (hunk 5 above)
Comment (davidmehren) [reply CONTRIBUTING.md]: done

**Review by mrdrogdrog — APPROVED** [review]

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,42 +1,67 @@
...
+
+1. Submit an issue describing your proposed change.
+   We will try to respond to your issue as soon as possible.
+2. Fork this repo, develop and test your code changes. Ensure you signed all your commits (see above for details).
+3. Submit a pull request against this repo's `master` branch.
+4. Your branch may be merged once all configured checks pass.
+
+[code-of-conduct]: ./CODE-OF-CONDUCT.md
+[community-forum]: https://community.hedgedoc.org
+[matrix-support]: https://matrix.to/#/#hedgedoc:matrix.org
```
Comment (ErikMichelson) [inline CONTRIBUTING.md:64]: `chat.hedgedoc.org` for easier possible future changes of the URL?

**Review by InnayTool — APPROVED** [review]

File: CONTRIBUTING.md (hunk 6 above)
Comment (davidmehren) [reply CONTRIBUTING.md]: done

**Review by ErikMichelson — APPROVED** [review]

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,42 +1,67 @@
...
+
+* Ensure the PR description is precise about the problem and your solution. Just fill out our template. That should cover the most important information.
+
+### Do you intend to add a new feature or change an existing one?
+
+* Suggest your idea via a new GitHub issue. After a confirmation about your idea, you can start writing code. Our maintainers and other project developers can provide useful details about the architecture and show you relevant issues and discussions.
+
+### Do you want to work on translations?
+
+* If you want to improve a translation or add a new translation altogether, we handle those via [POEditor](poeditor).
```
Comment (SISheogorath) [inline CONTRIBUTING.md:32]: This anchor need different brackets to work.

Comment (davidmehren) [reply CONTRIBUTING.md]: Fixed here and at all other places!

**Review by SISheogorath — APPROVED** [review]

**Pull request merged**